src/main/resources/spring/camel-context.xml

```diff
@@ -201,14 +216,20 @@
             </delete>
             <get uri="/{id}/initial-saving-estimation">
                 <description>Get the Initial Cost Saving report</description>
-                <route id="reports-get-details">
+                <route id="initial-saving-estimation-report-get-details">
                     <!-- RBAC check start-->
                     <setHeader headerName="${type:org.jboss.xavier.integrations.rbac.RBACRouteBuilder.RBAC_ENDPOINT_RESOURCE_NAME}"><constant>analysis.initialSavingEstimation</constant></setHeader>
                     <setHeader headerName="${type:org.jboss.xavier.integrations.rbac.RBACRouteBuilder.RBAC_ENDPOINT_RESOURCE_PERMISSION}"><constant>read</constant></setHeader>
                     <to uri="direct:check-rbac-permissions" />
                     <!-- RBAC check end-->
 
                     <bean ref="initialSavingsEstimationReportService" method="findByAnalysisOwnerAndAnalysisId(${header.${type:org.jboss.xavier.integrations.route.RouteBuilderExceptionHandler.USERNAME}}, ${header.id})" />
+                    <choice>
+                        <when>
+                            <simple>${body} == null</simple>
```

Comment:
@jonathanvila When we call `/report/{id}/initial-saving-estimation` there are 2 possible scenarios when `body == null` will be true:
1. When the `{id}` does not exist in the Database
2. When the `{id}` does not belong to the user (the report was created by another user).

My first thought is that we should return 2 different errors in each case; in the first case a `404` error and in the second case a `403 (forbidden)` error. However, this approach will require searching for reports regardless of who is the owner of it. What do you think? I'm ok returning `404` in the 2 scenarios I've described but I want to know your opinion.

Reply:
I think that we dont have to expose our internals..... if you are checking an Id you dont have access the return of the query will be null, therefore a 404 is what I think we have to return.